crates/core/src/db/queries/metadatas.rs

```diff
@@ -426,6 +429,164 @@ pub fn list(
         .context("Failed to load nft(s)")
 }
 
+/// Handles queries for a Collection Nfts
+///
+/// # Errors
+/// returns an error when the underlying queries throw an error
+#[allow(clippy::too_many_lines)]
+pub fn collection_nfts(
+    conn: &Connection,
+    collection: String,
+    auction_house: Option<String>,
+    attributes: Option<Vec<AttributeFilter>>,
+    marketplace_program: Option<String>,
+    sort_by: Option<Sort>,
+    order: Option<Order>,
+    limit: u64,
+    offset: u64,
+) -> Result<Vec<Nft>> {
+    let current_time = Utc::now().naive_utc();
+
+    let sort = match sort_by.unwrap() {
+        Sort::Price => Listings::Price,
+        Sort::ListedAt => Listings::CreatedAt,
+        _ => Listings::Price,
+    };
+
+    let mut listings_query = Query::select()
+        .columns(vec![
+            (Listings::Table, Listings::Metadata),
+            (Listings::Table, Listings::Price),
+            (Listings::Table, Listings::Seller),
+        ])
+        .from(Listings::Table)
+        .order_by((Listings::Table, sort), order.unwrap())
+        .cond_where(
+            Condition::all()
+                .add(Expr::tbl(Listings::Table, Listings::PurchaseId).is_null())
+                .add(Expr::tbl(Listings::Table, Listings::CanceledAt).is_null())
+                .add(
+                    Expr::tbl(Listings::Table, Listings::Expiry)
+                        .is_null()
+                        .or(Expr::tbl(Listings::Table, Listings::Expiry).gt(current_time)),
+                ),
+        )
+        .take();
```

Comment:
Just need a left join instead of lateral since only 1 marketplace can be selected at a time.